Sources/MongoSwift/ClientSession.swift

```diff
@@ -222,4 +222,52 @@ public final class ClientSession {
             }
         }
     }
+
+    /**
+     * Starts a multi-document transaction for all subsequent operations in this session. Any options provided in
+     * `options` override the default transaction options for this session and any options inherited from
```

Comment:
the mention of options precedence here made me wonder. what if I do the following:
- set no custom options on client
- for default txn options, set a custom read concern and write concern
- for options provided here, set a custom read concern, and do not specify a write concern in the options struct.

what is the write concern used for this txn? the custom one from the default options, or is it the default one since I didn't specify one in my options for this particular txn (yet did specify a read concern)?

I *think* consistency with our how we handle this inheritance in terms of DBs and collections inheriting from clients would mean that we should use the write concern from the default options: https://github.com/mongodb/mongo-swift-driver/blob/ec2bba2fa2779360a427695425d7589f93e9f843/Sources/MongoSwift/MongoClient.swift#L420

it may already work in that way but this comment made me realize we should probably verify that is the case.

Reply:
Yes, this is the case. The [transactions spec](https://github.com/mongodb/specifications/blob/master/source/transactions/transactions.rst#transactionoptions) explicitly state that "the writeConcern property is optional. The default value is NULL. If writeConcern is NULL the value will be inherited from this session’s defaultTransactionOptions. If defaultTransactionOptions itself is NULL or the writeConcern in defaultTransactionOptions is NULL, the writeConcern is inherited from the MongoClient associated with this session."

We set the default transactions options on the underlying `libmongoc` objects in `startTransaction`. I see that `MongoClient` does something similar. As such, `libmongoc` has all the necessary information to implement the logic described in the spec.